com.unity.render-pipelines.universal/Editor/AnimationClipUpgrader.cs

```diff
@@ -447,15 +447,29 @@ IEnumerable<ClipPath> clipPaths
                 // material property animations apply to all materials, so check shader usage in all of them
                 foreach (var material in rendererData.Materials)
                 {
-                    data.Usage |= UpgradeUtility.GetNewPropertyName(
+                    var usage = UpgradeUtility.GetNewPropertyName(
                         shaderProperty.Name,
                         material,
                         renameType,
                         allUpgradePathsToNewShaders,
                         upgradePathsUsedByMaterials,
                         out var newPropertyName
                     );
-                    data.PropertyRenames[shaderProperty.Name] = newPropertyName;
+
+                    // if the property has already been upgraded with a different name, mark the upgrade as ambiguous
+                    if (usage == SerializedShaderPropertyUsage.UsedByUpgraded)
+                    {
+                        if (data.PropertyRenames.TryGetValue(binding, out var propertyRename))
+                        {
+                            if (propertyRename != newPropertyName)
+                            {
+                                usage |= SerializedShaderPropertyUsage.UsedByAmbiguouslyUpgraded;
+                            }
+                        }
+                    }
```

Comment:
Minor nit, but I suppose this could be folded into a single conditional, right?
```suggestion
                    if (
                        usage == SerializedShaderPropertyUsage.UsedByUpgraded
                        && data.PropertyRenames.TryGetValue(binding, out var propertyRename)
                        && propertyRename != newPropertyName
                    )
                        usage |= SerializedShaderPropertyUsage.UsedByAmbiguouslyUpgraded;
```

Reply:
Works for me 👍